arrays/arrays.c

```diff
@@ -69,43 +80,61 @@ void resize_array(Array *arr) {
  *
  * Throw an error if the index is out of range.
  *****/
-char *arr_read(Array *arr, int index) {
+char *arr_read(Array *arr, int index)
+{
 
   // Throw an error if the index is greater than the current count
-
+  if (arr->count < index)
+  {
+    perror("Read error: index greater than current count");
+  }
   // Otherwise, return the element at the given index
+  return arr->elements[index];
 }
 
-
 /*****
  * Insert an element to the array at the given index
  *****/
-void arr_insert(Array *arr, char *element, int index) {
+void arr_insert(Array *arr, char *element, int index)
+{
 
   // Throw an error if the index is greater than the current count
-
+  if (arr->count < index)
+  {
+    perror("Insert error: index greater than current count");
+  }
   // Resize the array if the number of elements is over capacity
-
+  if (arr->count == arr->capacity)
+  {
+    Array *arr = resize_array(arr);
```

Comment:
If you take my advice above you won't need a new array and you can just call resize_array(arr) on this line.